Implement Edit Task prompt

kavisha-perera opened the pull request:
## Purpose
<!--- Describe the problems, issues, or needs driving this feature/fix and include links to related issues -->
The purpose of this PR is to fix part of https://github.com/we-huddle/server/issues/19 

## Goals
<!---  Describe the solutions that this feature/fix will introduce to resolve the problems described above -->
Created the Dummy Edit Task UI component 

## Approach
<!--- Describe how you are implementing the solutions. Include a link to a Markdown file or Google doc if the feature write-up is too long to paste here. -->

## Screenshots
<!--- Attach screenshots or demo-gif of any UI changes -->
![image](https://user-images.githubusercontent.com/92311289/185586361-51a01a41-82d5-4685-82aa-26a0ebe4f503.png)


## Related PRs
<!--- List any other related PRs --> 
https://github.com/we-huddle/client/pull/15

## Test environment
<!--- List all JDK versions, operating systems, databases, and browser/versions on which this feature/fix was tested --> 

## Learning
<!--- Describe the research phase and any blog posts, patterns, libraries, or add-ons you used to solve the problem. -->

kavisha-perera committed:
Implemented Dummy Edit Task UI

- Resolving merge conflicts

Gravewalker666 commented on src/scenes/TaskDetailsView/components/EditTaskPrompt/index.tsx:
```diff
@@ -0,0 +1,72 @@
+  );
+}
+
+export default EditTaskPrompt;
```
Put a new line at the end of the file (EOF)

kavisha-perera replied:
fixed

Gravewalker666 commented on src/scenes/TaskDetailsView/components/EditTaskPrompt/index.tsx:
```diff
@@ -0,0 +1,72 @@
+import {Button, Label, Modal, Textarea, TextInput} from "flowbite-react";
+
+interface EditTaskPromptProps {
+  show: boolean,
+  onClose: () => void,
+}
+
+function EditTaskPrompt({ show, onClose }: EditTaskPromptProps) {
```
Let's take the task as a property and populate the form with those data

Gravewalker666 replied:
This is a good working solution, well done.
We can further improve performance if we can somehow change this implementation from making another network call to getting the task passed down to the component via `props`. Let's edit the edit task prompt props to include the task in it. And then take that input parameter and render the form. 
Then you'll have to pass down the task object to this component from task details view.

Gravewalker666 commented on src/scenes/TaskDetailsView/index.tsx:
```diff
@@ -55,17 +67,19 @@ function TaskDetailsView({ isAgentView }: TaskDetailsViewProps){
-                <HiTrash className="ml-2" />
-              </Button></>
-            )}
+
+              {isAgentView && profile?.role === Profile.Role.HuddleAgent && (
+                  <>
+                  <Button onClick={() => setIsEditModalVisible(true)}>
+                      Edit
+                      <HiPencil className="ml-2" />
+                  </Button>
+                  <Button color="failure">
+                      Delete
+                      <HiTrash className="ml-2" />
+                  </Button></>
+              )}
+
```
```suggestion
              {isAgentView && profile?.role === Profile.Role.HuddleAgent && (
                  <>
                    <Button onClick={() => setIsEditModalVisible(true)}>
                        Edit
                        <HiPencil className="ml-2" />
                    </Button>
                    <Button color="failure">
                        Delete
                        <HiTrash className="ml-2" />
                    </Button>
                  </>
              )}
```
Not that important but better to have them properly formatted

kavisha-perera replied:
fixed

Gravewalker666 reviewed (changes requested):
Requested some changes

kavisha-perera committed:
implementing the requested changes in PR #19

kavisha-perera committed:
Implement the requested change in PR #19

including 'task' in EditTaskPrompt and rendering the form using that
input parameter.

Gravewalker666 reviewed (approved):
Great work @kavisha-perera thanks for getting this done
